tests/storage/datavolume.go

```diff
@@ -1461,10 +1498,14 @@ func newRandomVMWithCloneDataVolume(sourceNamespace, sourceName, targetNamespace
 		libdv.WithPVC(libdv.PVCWithStorageClass(sc), libdv.PVCWithVolumeSize("1Gi")),
 	)
 
-	vmi := tests.NewRandomVMIWithDataVolume(dataVolume.Name)
-	vmi.Namespace = targetNamespace
-	vm := libvmi.NewVirtualMachine(vmi)
+	vmi := libvmi.New(
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+		libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		libvmi.WithDataVolume("disk0", dataVolume.Name),
+		libvmi.WithResourceMemory("1Gi"),
+		libvmi.WithNamespace(targetNamespace),
+	)
 
-	libstorage.AddDataVolumeTemplate(vm, dataVolume)
+	vm := libvmi.NewVirtualMachine(vmi, libvmi.WithDataVolumeTemplate(dataVolume))
```

Comment:
You inlined the call to `libvmi.New` above, do it here too?

Reply:
Done